File: backend/app/models/schemas/events.py
from pydantic import BaseModel
from typing import Optional


class EventOut(BaseModel):
    id: int
    session_id: str
    frame_index: int
    timestamp_ms: int
    event_type: str
    severity: Optional[str]
    event_code: str
    from_state: Optional[str]
    to_state: Optional[str]
    trigger_metric: Optional[str]
    trigger_value: Optional[float]
    message: str
    metric_snapshot: Optional[str]
    telegram_sent: int

    model_config = {"from_attributes": True}

File: frontend/src/types/events.ts
export interface GrillEvent {
  id: number
  session_id: string
  frame_index: number
  timestamp_ms: number
  event_type: 'STATE_CHANGE' | 'ALERT'
  severity: 'WARNING' | 'CRITICAL' | null
  event_code: string
  from_state: string | null
  to_state: string | null
  trigger_metric: string | null
  trigger_value: number | null
  message: string
  metric_snapshot: string | null
  telegram_sent: number
}

export interface AlertItem {
  id: string
  code: string
  severity: 'WARNING' | 'CRITICAL'
  message: string
  timestamp: number
  frame_index?: number
  telegram_sent?: boolean
}
